src/ngraph/frontend/onnxifi/backend_manager.cpp

```diff
@@ -0,0 +1,67 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstdlib>   // std::size_t, std::uintptr_t
+#include <stdexcept> // std::invalid_agrument, std::out_of_rage
+
+#include <onnxifi.h>
+
+#include "ngraph/runtime/backend_manager.hpp"
+
+#include "backend.hpp"
+#include "backend_manager.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        BackendManager::BackendManager()
```

Comment:
If you need a special class rather than runtime::BackendManager then you also need a comment explaining why.

Reply:
`runtime::BackendManager::create(const std::string& type) -> std::unique_ptr<runtime::Backend>`

The most attractive feature of `std::unique_ptr` is that it easily and effectively converts to a `std::shared_ptr`. The backward conversion (`std::shared_ptr` to `std::unique_ptr`) is not allowed by design.

90% of nGraph code shoud be `std::unique_ptr` and ownership passing.